magenta-lib/src/main/scala/magenta/tasks/UpdateCloudFormationTask.scala

```diff
@@ -78,6 +79,19 @@ object UpdateCloudFormationTask {
         orderedTags.map{ case (key, value) => value }.mkString("-")
     }
   }
+
+  def processTemplate(stackName: String, templateBody: String, s3Client: AmazonS3, stsClient: AWSSecurityTokenServiceClient, region: Region): Template = {
+    val templateTooBigForSdkUpload = templateBody.length > 51200
+
+    if (templateTooBigForSdkUpload) {
+      val bucketName = S3.accountSpecificBucket("riff-raff-cfn-templates", s3Client, stsClient, region, Some(1))
+      val keyName = s"$stackName-${new DateTime().getMillis}"
+      s3Client.putObject(bucketName, keyName, templateBody)
+      TemplateUrl(s"s3://$bucketName/$keyName")
+    } else {
+      TemplateBody(templateBody)
+    }
```

Comment:
On second thoughts I've changed it so that if you are using the new riff-raff.yaml format then we'll upload by default. This seems like a good compromise - as people switch over they'll pick up the new behaviour without it breaking existing stuff.